routers/web/repo/contributors.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"errors"
+	"net/http"
+
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	contributors_service "code.gitea.io/gitea/services/repository"
+)
+
+const (
+	tplContributors base.TplName = "repo/activity"
+)
+
+// Contributors render the page to show repository contributors graph
+func Contributors(ctx *context.Context) {
+	ctx.Data["Title"] = ctx.Tr("repo.contributors")
+
+	ctx.Data["PageIsActivity"] = true
+	ctx.Data["PageIsContributors"] = true
+
+	ctx.PageData["contributionType"] = "commits"
+
+	ctx.PageData["repoLink"] = ctx.Repo.RepoLink
+
+	ctx.HTML(http.StatusOK, tplContributors)
+}
+
+// ContributorsData renders JSON of contributors along with their weekly commit statistics
+func ContributorsData(ctx *context.Context) {
+	if contributorStats, err := contributors_service.GetContributorStats(ctx, ctx.Cache, ctx.Repo.Repository, ctx.Repo.CommitID); err != nil {
+		if errors.Is(err, contributors_service.ErrAwaitGeneration) {
+			ctx.Status(216)
+			return
```

Comment:
Can you suggest change here @delvh ? I tried returning JSON but couldn't make it work :/

Reply:
Not really, as in that case we need to know exactly what to return, so it will need a new type definition which I cannot define in a suggestion.
To keep the algorithm working, this result type should even be returned in the successful case, however in that case containing the resulting data.

Once the result type has been designed, you can send it using
```go
ctx.JSON(http.StatusOK, <instance of the result type>
```

Reply:
OK, I also thought the same and asked to see if there is a simple solution. Apparently not.

In successful case this method returns something like 
```
[
  {"email1": [...data1]}, 
  {"email2": [...data2]}, 
  {"total": [...totalData]}
]
```

so `total` key is always there. I will return something like `total: null` if we are still waiting for the result. Does that sound good?

Reply:
Yep, sounds good

Reply:
(although a better approach would be to offer the resulting data only as a variable inside the body instead of as the whole body)

Reply:
>offer the resulting data only as a variable inside the body instead of as the whole body

Yep, that would be better. Maybe @6543 wanna do that? :) (I am a little bit tired working on this PR so I want to make things as fast as possible :d)